File: app/src/main/java/com/absinthe/anywhere_/utils/timber/ThreadAwareDebugTree.java
package com.absinthe.anywhere_.utils.timber;

import androidx.annotation.NonNull;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import timber.log.Timber.DebugTree;

public class ThreadAwareDebugTree extends DebugTree {

    @Override
    protected void log(int priority, String tag, @NonNull String message, Throwable t) {
        if (tag != null) {
            String threadName = Thread.currentThread().getName();
            tag = "<" + threadName + "> " + tag;
        }
        super.log(priority, tag, message, t);
    }

    @Override
    protected @Nullable String createStackElementTag(@NotNull StackTraceElement element) {
        return super.createStackElementTag(element) + " (Line " + element.getLineNumber() + ")";
    }
}

File: app/src/main/java/com/absinthe/anywhere_/utils/timber/ThreadAwareDebugTree.kt
package com.absinthe.anywhere_.utils.timber

import timber.log.Timber.DebugTree

open class ThreadAwareDebugTree : DebugTree() {

    override fun createStackElementTag(element: StackTraceElement): String? {
        //日志显示行号
        return super.createStackElementTag(element) + " (Line ${element.lineNumber})"
    }
}